src/services/exchangeRate.ts

```diff
@@ -31,36 +30,69 @@ function assertString(
   }
 }
 
-function parse(string: string): IExchangeRateResult[] {
+function parse(string: string): ExchangeRateDaily[] {
   const data = new XMLParser({
     ignoreAttributes: false,
     isArray: () => true,
   }).parse(string);
-  const result: IExchangeRateResult[] = [];
+  const result: ExchangeRateDaily[] = [];
   const entries = data["gesmes:Envelope"][0]["Cube"][0]["Cube"];
   if (typeof entries !== "object") {
     throw new Error("Result data does not have the expected structure");
   }
 
   for (const current of entries) {
-    const time: string = current?.["@_time"]?.[0];
-    assertString(time, "time");
-    const rates = {} as any;
+    const date = current?.["@_time"]?.[0];
+    assertString(date, "date");
+    let rates: ExchangeRates = {
+      USD: 0,
+      JPY: 0,
+      BGN: 0,
+      CZK: 0,
+      DKK: 0,
+      GBP: 0,
+      HUF: 0,
+      PLN: 0,
+      RON: 0,
+      SEK: 0,
+      CHF: 0,
+      ISK: 0,
+      NOK: 0,
+      HRK: 0,
+      RUB: 0,
+      TRY: 0,
+      AUD: 0,
+      BRL: 0,
+      CAD: 0,
+      CNY: 0,
+      HKD: 0,
+      IDR: 0,
+      ILS: 0,
+      INR: 0,
+      KRW: 0,
+      MXN: 0,
+      MYR: 0,
+      NZD: 0,
+      PHP: 0,
+      SGD: 0,
+      THB: 0,
+      ZAR: 0,
```

Comment:
instead of this would {[key: ExchangeRates]: 0} since we already have all currencies in model?